### PR TITLE
Fixes #4471 - Don't call Enum.ToString()

### LINKED ISSUE #4471: GenerateForm calls Enum.ToString
Sampling data from BasicViews - percentages are relative to time spent in the view

image

Notice here that calling `Enum.ToString` and all of the things that it does are **more expensive** than generating the URL in routing. It's **half as expensive** as generating an antiforgery token, which does crypto....

### PR DESCRIPTION
**Pull request by rynowak**
Best perf fix ever :laughing:

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs [hunk 1]
```diff
@@ -229,7 +229,7 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                     output.PostContent.AppendHtml(tagBuilder.InnerHtml);
                 }
 
-                if (string.Equals(Method, FormMethod.Get.ToString(), StringComparison.OrdinalIgnoreCase))
+                if (string.Equals(Method, "get", StringComparison.OrdinalIgnoreCase))
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs]: But what if `FormMethod.Get.ToString` returns `post`?

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs]: That won't happen 'til c#9 I'm sure.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs [hunk 2]
```diff
@@ -1104,7 +1104,7 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
             if (string.IsNullOrEmpty(method))
             {
                 // Occurs only when called from a tag helper.
-                method = FormMethod.Post.ToString().ToLowerInvariant();
+                method = "post";
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs]: Suggest sticking "post" in a `const string` in this class.

**Comment by dougbu** [comment]
:shipit:

**Pull request closed without merging**